### PR TITLE
Added matchmaker loop and tested it

### PR DESCRIPTION
**Pull request by Demoncrush44**

### CODE REVIEW

File: matchmaker.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"fmt"
+	"time"
+)
+
+func matchUsers(db *DB) {
+	go func() {
+		for {
+			// Wait for the specified interval (2 seconds) before proceeding with matching users.
+			time.Sleep(2 * time.Second)
```
Comment (kphan32) [inline matchmaker.go:12]: [here's a better way of starting an interval loop](https://gobyexample.com/timers)

File: matchmaker.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			// Match users together to create sessions.
+			// pair users in the order they are in the list.
+			for i := 0; i < len(unmatchedUsers); i += 2 {
+				if i+1 < len(unmatchedUsers) {
+					// Pair two users and create a session for them.
+					session := db.NewSession(unmatchedUsers[i], unmatchedUsers[i+1])
+
+					// Send the session and other user ID to each user via SSE.
+					sendSessionToUser(unmatchedUsers[i], session.ID, unmatchedUsers[i+1].ID)
+					sendSessionToUser(unmatchedUsers[i+1], session.ID, unmatchedUsers[i].ID)
```
Comment (kphan32) [inline matchmaker.go:26]: i would store the matched users in variables, EG `userA := unmatchedUsers[i]`

File: main.go [hunk 3]
```diff
@@ -12,6 +12,9 @@ func main() {
 	// Create the database instance.
 	db := NewDB()
 
+	// Start the matching loop in a separate goroutine
+	matchUsers(db)
```
Comment (kphan32) [inline main.go:16]: rename this to something like `startMatchMakerLoop`

Comment (kphan32) [reply main.go:16]: also, this should be the function receiving the `go` keyword, EG `go startMatchMakerLoop()`

File: matchmaker.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+func matchUsers(db *DB) {
+	go func() {
```
Comment (kphan32) [inline matchmaker.go:9]: don't start the go routine in this function

**Review by kphan32 — CHANGES_REQUESTED** [review]

File: matchmaker.go (hunk 2 above)
Comment (Demoncrush44) [reply matchmaker.go:26]: Got it

File: matchmaker.go (hunk 4 above)
Comment (Demoncrush44) [reply matchmaker.go:9]: What would you recommend I should do Im still iffy on goroutines

File: matchmaker.go (hunk 1 above)
Comment (Demoncrush44) [reply matchmaker.go:12]: On it

**Review by kphan32 — APPROVED** [review]
lgtm :+1:

**Pull request merged**